lib/ex_aws/parsers.ex

```diff
@@ -0,0 +1,18 @@
+defmodule ExAws.Parsers do
+  defmacro __using__(_opts) do
+    quote do
+      import SweetXml, only: [sigil_x: 2]
+      def parse({:error, {type, http_status_code, %{body: xml}}}, _) do
+        parsed_body = xml
+        |> SweetXml.xpath(~x"//ErrorResponse",
+                          request_id: ~x"./RequestId/text()"s,
+                          type: ~x"./Error/Type/text()"s,
+                          code: ~x"./Error/Code/text()"s,
+                          message: ~x"./Error/Message/text()"s,
+                          detail: ~x"./Error/Detail/text()"s)
```

Comment:
Fair enough. I think it should likely go under `ExAws.Operation.Query.Parser` because the parsing it does seems specific to that particular operation format. It does not relate to either S3 or say Dynamo.